workspaces/orchestrator/plugins/orchestrator/src/components/WorkflowPage/WorkflowPageTabContent.tsx

```diff
@@ -39,11 +43,16 @@ export const WorkflowPageTabContent = ({
   canRun,
   children,
 }: Props) => {
-  const { workflowId } = useRouteRefParams(workflowRouteRef);
+  const { workflowId, format } = useRouteRefParams(workflowRouteRef);
+  const definitionRunsLink = useRouteRef(workflowRunsRouteRef);
   const navigate = useNavigate();
   const executeWorkflowLink = useRouteRef(executeWorkflowRouteRef);
   const handleExecute = () => {
-    navigate(executeWorkflowLink({ workflowId }));
+    navigate(executeWorkflowLink({ workflowId }), {
+      state: {
+        from: definitionRunsLink({ workflowId: workflowId, format: format }),
```

Comment:
I might be wrong but maybe we do not need to recompose the URL and use `location.pathname` instead.

Reply:
It's possible but I'll have to slice the "/orchestrator" from the beginning of `location.pathname`